### PR TITLE
[Milvus] connector improvements

### PR DESCRIPTION
**Pull request by EugeneLightsOn**
### What's being changed:
The Milvus connector refactoring and improvements.

### How did you test this change (include any code snippets, API requests, screenshots, or gifs):
image

### CODE REVIEW

File: milvus/dev/load_data.py [hunk 1]
```diff
@@ -12,19 +12,26 @@
...
+    if "MILVUS_API_KEY" not in os.environ or os.environ["MILVUS_API_KEY"] == "":
+        connections.connect(
+            alias="default",
+            uri=os.environ["MILVUS_CLUSTER_URI"],
+        )
+    else:
+        connections.connect(
+            alias="default",
+            uri=os.environ["MILVUS_CLUSTER_URI"],
+            token=os.environ["MILVUS_API_KEY"],
```
Comment (tianjing-li) [inline milvus/dev/load_data.py:24]: Out of curiosity is it ok to connect without an API key? Does the functionality change

File: milvus/provider/client.py [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        api_key=None,
+    ):
+        if api_key is not None:
+            connections.connect(
+                alias=self.DEFAULT_ALIAS, uri=cluster_uri, token=api_key
+            )
+        else:
+            connections.connect(
+                alias=self.DEFAULT_ALIAS, uri=cluster_uri, token=f"{user}:{password}"
+            )
```
[the remark this answers is not in the document]
Comment (tianjing-li) [reply milvus/provider/client.py:31]: After checking `elif user != "" and password != ""`, in an else

File: milvus/provider/client.py [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+        self.collection = Collection(name=collection_name)
+        self.vector_field = vector_field
+        self.search_limit = self.DEFAULT_SEARCH_LIMIT
+
+    def set_search_limit(self, search_limit):
```
Comment (tianjing-li) [inline milvus/provider/client.py:36]: I would remove the setter, either you can pass the search_limit into the `__init__` method or just set the search_limit attribute directly, in general setting/getting in Python is not recommended unless you have specific reasons to do so

**Review by tianjing-li — APPROVED** [review]
Couple minor comments but looks good

File: milvus/README.md [hunk 4]
```diff
@@ -2,19 +2,79 @@
 
 This package is a utility for connecting Cohere to a Milvus database.
 
-It relies on the `pymilvus` package for managing establishing the connection and performing vector searches. This implementation also uses Cohere's embedding API to generate search vectors.
+It relies on the `pymilvus` package for managing establishing the connection and performing vector searches.
```
Comment (daniel-cohere) [inline milvus/README.md:5]:
```suggestion
It relies on the `pymilvus` package for establishing/managing the connection and performing vector searches. 
```

File: milvus/README.md [hunk 5]
```diff
@@ -2,19 +2,79 @@
...
 
-It relies on the `pymilvus` package for managing establishing the connection and performing vector searches. This implementation also uses Cohere's embedding API to generate search vectors.
+It relies on the `pymilvus` package for managing establishing the connection and performing vector searches. 
+This implementation also uses Cohere's embedding API to generate search vectors.
 
 ## Limitations
 
-Currently the search is performed by embedding the search query with Cohere's embedding API.
+Currently, the search is performed by embedding the search query with Cohere's embedding API.
+So the Milvus DB vectors embeddings should be generated using Cohere embed API.
```
Comment (daniel-cohere) [inline milvus/README.md:11]:
```suggestion
So the Milvus DB vectors embeddings should be generated using the Cohere embed API.
```

File: milvus/provider/provider.py [hunk 6]
```diff
@@ -1,62 +1,52 @@
...
+    """
+    serialized_data = list(
+        map(
+            lambda item: {
+                k
+                if k not in mappings
+                else mappings[k]: ", ".join(str(vl) for vl in v)
+                if isinstance(v, list)
+                else str(v)
+                for k, v in item.items()
```
Comment (daniel-cohere) [inline milvus/provider/provider.py:29]: Could we break this out into a proper function instead of a lambda? Finding the `if ... else` hard to follow here

**Review by daniel-cohere — APPROVED** [review]

File: milvus/dev/load_data.py (hunk 1 above)
Comment (EugeneLightsOn) [reply milvus/dev/load_data.py:24]: The load data script can be used for both standalone and managed clusters. The standalone cluster allows connection without a key and username/password.

File: milvus/provider/client.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (EugeneLightsOn) [reply milvus/provider/client.py]: For the standalone instance, connection is possible without a username/password and without an API key. For the managed instance (for example, https://cloud.zilliz.com/), either a username/password or an API key can be used to log in to the cluster. A minor fix has been applied to handle exceptions more gracefully.

File: milvus/provider/client.py (hunk 3 above)
Comment (EugeneLightsOn) [reply milvus/provider/client.py]: Thank you. Done

File: milvus/README.md (hunk 5 above)
Comment (EugeneLightsOn) [reply milvus/README.md]: Thank you

File: milvus/provider/provider.py (hunk 6 above)
Comment (EugeneLightsOn) [reply milvus/provider/provider.py]: Thank you. Done
